core/database/tests/test_foxx.sh

```diff
@@ -128,6 +128,14 @@ if ! command -v foxx > /dev/null 2>&1; then
 fi
 
 PATH_TO_PASSWD_FILE=${SOURCE}/database_temp.password
+
+# set up test fixtures, this script should be idempotent, this script is described in the manifest
+"${FOXX_PREFIX}foxx" script -u "${local_DATABASE_USER}" \
+   --server "tcp://${DATAFED_DATABASE_HOST}:8529" \
+    -p "${PATH_TO_PASSWD_FILE}" \
+    --database "${local_DATABASE_NAME}" \
+    "/api/${local_FOXX_MAJOR_API_VERSION}" user-fixture
+
 if [ "$TEST_TO_RUN" == "all" ]
```

Comment:
CMake already provides a means for doing this take a look here.

https://github.com/ORNL/DataFed/blob/ebef1b003a5b270ae11023a0aadf8c73f0726404/core/database/CMakeLists.txt#L1C1-L29C8

If you want to only make these changes for this test. You can add a separte cmake setup fixture. Otherwise, you can add it directly in https://github.com/ORNL/DataFed/blob/1180-refactor-authz/core/database/tests/test_setup.sh, 

Best practice would probably be to keep it separate. And add something like this to the CMakeLists.txt file. You will need a separate shell script which should probably be placed in here. https://github.com/ORNL/DataFed/tree/1180-refactor-authz/core/database/tests

```
add_test(NAME foxx_setup_user COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/tests/test_setup_user.sh")
add_test(NAME foxx_user COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/tests/test_foxx.sh" -t "unit_user")

set_tests_properties(foxx_setup PROPERTIES FIXTURES_SETUP FoxxUser)
set_tests_properties(foxx_setup_user PROPERTIES FIXTURES_SETUP FoxxUser DEPENDS foxx_setup)
set_tests_properties(foxx_teardown PROPERTIES FIXTURES_CLEANUP FoxxUser)
set_tests_properties(foxx_user PROPERTIES FIXTURES_REQUIRED FoxxUser)
```

Might want to take a look at this documentation. 

https://cmake.org/cmake/help/latest/prop_test/FIXTURES_SETUP.html

/core/database/tests/test_setup_user.sh - this is incomplete
```
"${FOXX_PREFIX}foxx" script -u "${local_DATABASE_USER}" \
   --server "tcp://${DATAFED_DATABASE_HOST}:8529" \
    -p "${PATH_TO_PASSWD_FILE}" \
    --database "${local_DATABASE_NAME}" \
    "/api/${local_FOXX_MAJOR_API_VERSION}" user-fixture
```